### PR TITLE
WT-7928 VLCS checkpoint and additional test suite improvements

### PR DESCRIPTION
**Pull request by sauclovian-wt**
Found a column-store bug that was not captured by the Python tests. The accompanying changeset adds a Python test for that problem, and contains additional improvements to both the C and Python test suites; mostly further or re-enabled column-store test cases, but also fixes test_checkpoint and enables previously-broken row-store cases as well.

### CODE REVIEW

**Comment by lukech** [comment]
autobuild: Can one of the admins verify this patch?

**Comment by kommiharibabu** [comment]
@sauclovian-wt Shall we enable the column store tests to run as part of evergreen `test/evergreen.yml` in this PR?

**Comment by sauclovian-wt** [comment]
@kommiharibabu Seems reasonable. I didn't want to venture it myself since I have nearly zero ability to see the results...

(Even with this fix there's still at least one problem left that test_format chokes on, though so it's not time yet to turn that one on.)

also note that this pull request needs to hold until WT-7908 gets merged again.

**Comment by sauclovian-wt** [comment]
also: when that happens, it's ticket WT-6241

**Comment by keithbostic** [comment]
@kommiharibabu, @quchenhao, this one is ready to go, and I've flagged you as the reviewers. Please feel free to tag someone else you think would be better choice, of course.

**Review by quchenhao — APPROVED** [review]
lgtm

File: test/csuite/random_abort/main.c [hunk 1]
```diff
@@ -162,15 +162,11 @@ thread_run(void *arg)
 
     testutil_check(td->conn->open_session(td->conn, NULL, NULL, &session));
 
-#if 0
     /*
      * Make sure that alternative threads operate on column-store table
```
Comment (quchenhao) [inline test/csuite/random_abort/main.c:166]: Can change to single line comment format.

File: test/suite/test_bug022.py [hunk 2]
```diff
@@ -30,27 +30,37 @@
...
 import wiredtiger, wttest
+from wtscenario import make_scenarios
 
 class test_bug022(wttest.WiredTigerTestCase):
     uri = 'file:test_bug022'
     conn_config = 'cache_size=50MB'
     session_config = 'isolation=snapshot'
 
+    key_format_values = [
+        ('string', dict(key_format='S', usestrings=True)),
```
Comment (kommiharibabu) [inline test/suite/test_bug022.py:41]: string -> string-row? a similar change in all the other tests.

File: test/suite/test_durable_ts03.py [hunk 3]
```diff
@@ -28,18 +28,25 @@
...
+from wtscenario import make_scenarios
 
 # test_durable_ts03.py
 #    Check that the checkpoint honors the durable timestamp of updates.
 class test_durable_ts03(wttest.WiredTigerTestCase):
     conn_config = 'cache_size=10MB'
     session_config = 'isolation=snapshot'
 
+    key_format_values = [
+        ('int', dict(key_format='i')),
```
Comment (kommiharibabu) [inline test/suite/test_durable_ts03.py:40]: int -> integer-row? a similar change in all the tests.

File: test/csuite/random_abort/main.c (hunk 1 above)
Comment (sauclovian-wt) [reply test/csuite/random_abort/main.c]: Fixed

File: test/suite/test_bug022.py (hunk 2 above)
Comment (sauclovian-wt) [reply test/suite/test_bug022.py]: Fixed these, thanks. This lot was sitting on a different git branch and so got missed by the previous massedit.

**Review by kommiharibabu — APPROVED** [review]
LGTM. Thanks.

**Comment by sauclovian-wt** [comment]
That's the same problem the other ticket's branch had... oh, I think the change to temporarily disable that test isn't in this branch. Guess I should merge...

**Pull request merged**